cmake/Modules/FindGloo.cmake

```diff
@@ -1,45 +1,22 @@
-# Try to find the Gloo library and headers.
-#  Gloo_FOUND        - system has Gloo lib
-#  Gloo_INCLUDE_DIRS - the Gloo include directory
-#  Gloo_LIBRARIES    - libraries needed to use Gloo
+IF(NOT GLOO_FOUND)
```

Comment:
Looks like the two FindGloo.cmake histories are

Caffe2 - single commit May 2017
https://github.com/pytorch/pytorch/commits/a2a28c0ef1d9a433972fe72fa5b0b9b850ccfcaf/cmake/Modules/FindGloo.cmake

PyTorch - a couple of commits March - May 2017
https://github.com/pytorch/pytorch/commits/4ebf3ff46dc0d4899fb163a970ba4dc68c566b2a/torch/lib/THD/cmake/FindGloo.cmake

@apaszke and @pietern seemed to have touched these. Which do you think should win and/or how should we merge?

Reply:
In this case I'm going to take the Caffe2 version. @apaszke let me know if you think otherwise.

Reply:
The Caffe2 looks nicer to me, so I think we should go with that one.